### PR TITLE
mixedversion: remove special case for skipping current version

### PR DESCRIPTION
**Pull request by DarrylWong**
In the past, we used to bump CRDB's minSupportedVersion _before_ bumping CRDB's current version.

Consider the old version bump process:
  1. Bump minimum supported version (e.g. 25.2→25.3)
  2. Bump current version (e.g. 25.4→26.1)
  3. Bump minimum supported version again (e.g. 25.3→25.4)

After the first step, we would be in a state where the mixed version framework would see v25.3 as a skippable version (.1 and .3 ordinal versions are skippable), but since the minimum supported version is 25.3, 25.2→25.4 is (temporarily) not a legal upgrade.

This required a special case to disallow skip upgrades in the case of this edge case. However, we now bump the current version _first_. This means we no longer need to protect against the above and we actually hit a different bug.

Since we now bump the current version first, the special case will always view the current version as having multiple supported predecessors and force a skip upgrade even if not applicable.

Release note: none
Epic: none

### CODE REVIEW

**Review by williamchoe3 — APPROVED** [review]
Saw the thread in #test-eng too, changes make sense, skimmed the surrounding logic to understand the edge case, lgtm. This type of thing seems to be a consequence of the messy release support strategy / versioning of innovation and non innovation releases

File: pkg/cmd/roachtest/roachtestutil/mixedversion/mixedversion.go [hunk 1]
```diff
@@ -613,14 +611,6 @@ func (t *Test) supportsSkipUpgradeTo(pred, v *clusterupgrade.Version) bool {
 	if t.options.minimumSupportedVersion.Series() == pred.Series() {
 		return false
 	}
-	// Special case for the current release series. This is useful to keep the
-	// test correct when we bump the minimum supported version separately from
-	// the current version.
-	r := clusterversion.Latest.ReleaseSeries()
-	currentMajor := version.MajorVersion{Year: int(r.Major), Ordinal: int(r.Minor)}
```
Comment (srosenberg) [inline pkg/cmd/roachtest/roachtestutil/mixedversion/mixedversion.go:620]: The fact that `Test_UpgradePaths` didn't fail suggests it's weakly specified. I think we should strengthen it by asserting the "ordinal invariant", i.e., no skips from even.
